### PR TITLE
Feat : Add Note 

### PR DESCRIPTION
**Pull request by dikshadhote**
The following features have been added:

- Add label, priority ,color , and pin message while creating card
image

- Display card created in UI
image

PS: Find deploy preview in netlify bot

### CODE REVIEW

File: src/ApiServices/ApiServices.js [hunk 1]
```diff
@@ -26,4 +28,38 @@ const logInHandler = async ({ email, password }) => {
...
 
-export { signUpHandler, logInHandler };
+// Notes
+
+const createNoteHandler = async (data) => {
+  try {
+    const res = await axios.post(
+      "/api/notes",
+      {
+        note: { ...data },
```
Comment (tusharkandpal) [inline src/ApiServices/ApiServices.js]: i guess here spread operator is not needed.

File: src/Context/notes-context.js [hunk 2]
```diff
@@ -57,8 +73,11 @@ const NoteProvider = ({ children }) => {
     priority: "",
   });
 
+  useEffect(() => {
+    getData();
+  }, [noteState]);
```
Comment (tusharkandpal) [inline src/Context/notes-context.js:78]: its better to getData at the time of Login, not everytime the noteState is changed.

File: src/components/Home/DisplayNote.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            >
+              <div className="d-flex flex-justify-space-between  align-items-center p-1">
+                <h5>{title}</h5>
+                <BsPinAngleFill
+                  className={
+                    note.isPin
+                      ? "aqua-color cursor-pointer"
+                      : "white-text-color cursor-pointer"
+                  }
+                  title={isPin ? "Unpin this note" : "Pin this note"}
```
Comment (tusharkandpal) [inline src/components/Home/DisplayNote.js:41]: liked that you're setting title too. !

**Review by tusharkandpal — APPROVED** [review]
well organized the files...even the colors are separate. Overall, good work.

on hosted site, i found few observations:
- the first note is setting as empty even after writing content, maybe the state issue.
- also, blank values are getting accepted.

One suggestion: can use NavLink to show the active link in Sidebar

**Review by atharvabhanage02 — APPROVED** [review]
Code is well structured and `Add Note` functionality is working fine.

File: src/Context/notes-context.js [hunk 4]
```diff
@@ -57,8 +73,11 @@ const NoteProvider = ({ children }) => {
     priority: "",
   });
 
+  useEffect(() => {
```
Comment (ankit9015) [inline src/Context/notes-context.js:76]: Calling getNotes at the time of page load is enough

File: src/components/Home/CreateNote.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+      >
+        <div className="d-flex flex-justify-space-between align-items-center">
+          <input
+            className={
+              noteState.cardColor.inputColor
+                ? `input-create-note gray-text-note  fs-2 ${noteState.cardColor.inputColor}`
+                : "input-create-note gray-text-note fs-2 black-light-bg"
+            }
+            placeholder="Title"
+            onChange={(e) => {
```
Comment (ankit9015) [inline src/components/Home/CreateNote.js:42]: I like that you are using timer to call dispatch

File: src/components/Home/DisplayNote.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                  className={priority ? "chips black-text-color" : "hide-label"}
+                >
+                  {priority}
+                </span>
+              </div>
+              <div className="d-flex flex-justify-around mt-1 align-items-center">
+                <div className="pos-relative">
+                  <MdOutlineLabel
+                    className="fs-3 white-text-color cursor-pointer"
+                    title="Add label"
```
Comment (ankit9015) [inline src/components/Home/DisplayNote.js:66]: I like that you are setting title for the buttons

**Review by ankit9015 — APPROVED** [review]
PR is good to merge

**Comment by ankit9015** [comment]
The Note Editor is not going back to an empty state after a note is saved, same for labels. You should try to put them back to empty state, once their function is done

**Comment by dikshadhote** [comment]
> well organized the files...even the colors are separate. Overall, good work.
> 
> on hosted site, i found few observations:
> 
> * the first note is setting as empty even after writing content, maybe the state issue.
> * also, blank values are getting accepted.
> 
> One suggestion: can use NavLink to show the active link in Sidebar

Okay will remove timer, empty content will be because of that

**Comment by dikshadhote** [comment]
> The Note Editor is not going back to an empty state after a note is saved, same for labels. You should try to put them back to empty state, once their function is done

Yes, Currently this functionality is not available, will be available in future

File: src/Context/notes-context.js (hunk 2 above)
Comment (dikshadhote) [reply src/Context/notes-context.js:78]: User will logged in once, and I want to get data when state is updated as the color , pin status can be changed

File: src/ApiServices/ApiServices.js (hunk 1 above)
Comment (dikshadhote) [reply src/ApiServices/ApiServices.js]: Okay removed

File: src/Context/notes-context.js (hunk 2 above)
Comment (tusharkandpal) [reply src/Context/notes-context.js:78]: alright.. but you're getting an array of notes in response too.

**Pull request merged**